File: Problem1/dcp1.py
#!/user/bin/env python3 -tt
"""
Module documentation.
"""

# Imports
import sys
#import os

# Global variables

# Class declarations

# Function declarations
def sumExists(lst , k):
    seen = set()
    for num in lst:
        d = k - num
        if d in seen:
            return True
        else:
            seen.add(num)
    return False





def main():
    print(sumExists([10,15,3,7], 17))
    print(sumExists([10,15,3,4], 17))
    print(sumExists([10,15,3,7,4,8,9], 12))
    print(sumExists([10,15,3,7,8], 12))

# Main body
if __name__ == '__main__':
    main()

File: Problem1/dcp1.cpp
#include <iostream>
#include <vector>

using namespace std;


bool sumExists(vector<int> lst, int k){
  int d;
  vector<int> seen (k , 0);
  for(vector<int>::iterator it = lst.begin(); it != lst.end(); ++it){
    d = k - *it;
    if(d <= k){
      if(seen[d] == 0){
        seen[*it] = 1;
      }
      else{
        return true;
      }
    }
  }
  return false;
}


int main()
{
    vector<int> lst;
    lst.push_back (10);
    lst.push_back (15);
    lst.push_back (3);
    lst.push_back (7);
    bool answer = sumExists(lst , 17);
    cout << answer;

    return 0;
}
